stem/output.py

```diff
@@ -0,0 +1,219 @@
+    VELOCITY_Y = 22
+    VELOCITY_Z = 23
+    ACCELERATION = 3
+    ACCELERATION_X = 31
+    ACCELERATION_Y = 32
+    ACCELERATION_Z = 33
+    TOTAL_DISPLACEMENT = 4
+    TOTAL_DISPLACEMENT_X = 41
+    TOTAL_DISPLACEMENT_Y = 42
+    TOTAL_DISPLACEMENT_Z = 43
+    WATER_PRESSURE = 5
+    VOLUME_ACCELERATION = 6
+    VOLUME_ACCELERATION_X = 61
+    VOLUME_ACCELERATION_Y = 62
+    VOLUME_ACCELERATION_Z = 63
+
+
+TENSOR_OUTPUTS = [
+    "GREEN_LAGRANGE_STRAIN",
+    "ENGINEERING_STRAIN",
+    "CAUCHY_STRESS",
+    "TOTAL_STRESS",
+]
+
+
+def detect_tensor_outputs(requested_outputs: List[str]):
+    """
+    Detects whether gauss point outputs are requested and warns the user
+
+    Args:
+        requested_outputs (List[st]): list of requested outputs (gauss point)
+    """
+    if len(requested_outputs) > 0:
+        detected_tensor_outputs = []
+        for requested_output in requested_outputs:
+            for tensor_output in TENSOR_OUTPUTS:
+                if tensor_output in requested_output:
+                    detected_tensor_outputs.append(tensor_output)
+                    break
+        if len(detected_tensor_outputs):
+            _fmt_list = "".join([f" - {outpt} \n" for outpt in detected_tensor_outputs])
+            _msg = (
+                f"\n[WARNING] tensor output detected:\n{_fmt_list}"
+                f"The outputs are incorrectly rendered for the selected output\n"
+                f"if vector is selected and ignored if tensor is selected."
+            )
+            print(_msg)
+
+
+@dataclass
+class OutputParametersABC(ABC):
+    """
+    Abstract class for the definition of user output parameters (GiD, VTK, json).
+
+    Attributes:
+        -
+    """
+
+    pass
+
+    @abstractmethod
+    def validate(self):
+        """
+        Abstract method for validating user inputs
+        """
+        pass
+
+
+@dataclass
+class GiDOutputParameters(OutputParametersABC):
+    """
+    Class containing the output parameters for GiD output
+
+    Attributes:
+        gidpost_flags (Dict[str, Any]):
+        file_label (str):
+        output_control_type (str):
+        output_interval (int):
+        body_output (bool):
+        node_output (bool):
+        skin_output (bool):
+        plane_output (List[str]):
+        nodal_results (List[str]):
+        gauss_point_results (List[str]):
+        point_data_configuration (List[str]):
+    """
+
+    nodal_results: List[str] = field(default_factory=lambda: [])
+    gauss_point_results: List[str] = field(default_factory=lambda: [])
+
+    gid_post_mode: str = "GiD_PostBinary"
+    file_label: str = "step"
+    output_control_type: str = "step"
+    output_interval: int = 1
+    body_output: bool = True
+    node_output: bool = False
+    skin_output: bool = False
+    # optional
+    plane_output: List[str] = field(default_factory=lambda: [])
+    point_data_configuration: List[str] = field(default_factory=lambda: [])
+
+    def validate(self):
+        pass
```

Comment:
I remember that vector output on gauss points gave problems on gid, can you check this and validate?

Reply:
That is correct, it gives undefined. Should be checked for tensor output + _VECTOR specified.